modules/extensions-xml-support/src/test/resources/modules/module-tls-config-with-default.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module name="module-tls-config-with-default"
+        xmlns="http://www.mulesoft.org/schema/mule/module"
+        xmlns:petstore="http://www.mulesoft.org/schema/mule/petstore"
+        xmlns:tls="http://www.mulesoft.org/schema/mule/tls"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="
+           http://www.mulesoft.org/schema/mule/module http://www.mulesoft.org/schema/mule/module/current/mule-module.xsd
+           http://www.mulesoft.org/schema/mule/tls http://www.mulesoft.org/schema/mule/tls/current/mule-tls.xsd
+           http://www.mulesoft.org/schema/mule/petstore http://www.mulesoft.org/schema/mule/petstore/current/mule-petstore.xsd ">
+
+    <property name="someUserConfig" type="string"/>
+    <property name="somePassConfig" type="string"/>
+
+    <petstore:config name="petstore-config" cashierName="cashierName">
+        <petstore:connection username="#[vars.someUserConfig]" password="#[vars.somePassConfig]" xmlns:tlsEnabled="true">
+            <tls:context>
```

Comment:
so the semantics is that if no TLS is provided this one is used by default? Does this mean the tls parameter will always be optional?

Reply:
Yes, we will always be declaring the new synthetic parameter as optional (see [here](https://github.com/mulesoft/mule/pull/12952/files/701011683b2cff91c81866072b15673267810d8b#diff-2171aea98cf85c8e6bc80674bdeb462f3d559c2d8e931e48c7c1ae0fb79f293aR93)).

So, at least from the point of view of the XML module, it is always optional.

Now, I'm thinking there might be a case in which the target element didn't have the `TlsContextFactory` parameter as optional. Right now, such situation will force the developer of the XML module to provide a default TLS context, otherwise it will fail when validating the module's AST because it is missing a required parameter (previous macro-expansion). I need to check that.

Reply:
make sure to reflect this in the ADR